File: conformance/tests/arithmetic/arithmetic.tsop.go
package contract

import "tsop"

type Arithmetic struct {
	tsop.SmartContract
	Target tsop.Int `tsop:"readonly"`
}

func (c *Arithmetic) Verify(a tsop.Int, b tsop.Int) {
	sum := a + b
	diff := a - b
	prod := a * b
	quot := a / b
	result := sum + diff + prod + quot
	tsop.Assert(result == c.Target)
}

File: conformance/tests/arithmetic/arithmetic.tsop.rs
use tsop::prelude::*;

#[tsop::contract]
struct Arithmetic {
    #[readonly]
    target: Int,
}

#[tsop::methods(Arithmetic)]
impl Arithmetic {
    #[public]
    fn verify(&self, a: Int, b: Int) {
        let sum = a + b;
        let diff = a - b;
        let prod = a * b;
        let quot = a / b;
        let result = sum + diff + prod + quot;
        assert!(result == self.target);
    }
}
